myst_nbs/time_series/Forecasting_with_structural_timeseries.myst.md

```diff
@@ -0,0 +1,747 @@
+            constant=True,
+            steps=t.shape[0] - (priors["coefs"]["size"] - 1),
+        )
+
+        ## Priors for the linear trend component
+        alpha = pm.Normal("alpha", priors["alpha"]["mu"], priors["alpha"]["sigma"])
+        beta = pm.Normal("beta", priors["beta"]["mu"], priors["beta"]["sigma"])
+        trend = pm.Deterministic("trend", alpha + beta * t, dims="obs_id")
+
+        mu = ar1 + trend
+
+        # The Likelihood
+        outcome = pm.Normal("likelihood", mu=mu, sigma=sigma, observed=y)
+        ## Sampling
+        idata_ar = pm.sample_prior_predictive()
+        if full_sample:
+            idata_ar.extend(pm.sample(samples, random_seed=100, target_accept=0.95))
+            idata_ar.extend(pm.sample_posterior_predictive(idata_ar))
+        else:
+            return idata_ar
+
+    n = prediction_steps - ar_data.shape[0]
+
+    with AR:
+        AR.add_coords({"obs_id_fut": range(ar1_data.shape[0], prediction_steps, 1)})
+        t_fut = pm.MutableData("t_fut", list(range(ar1_data.shape[0], prediction_steps, 1)))
+        # condition on the learned values of the AR process
+        # initialise the future AR process precisely at the last observed value in the AR process
+        # using the special feature of the dirac delta distribution to be 0 probability everywhere else.
+        ar1_fut = pm.AR(
+            "ar1_fut",
+            init_dist=pm.DiracDelta.dist(ar1[..., -1]),
+            rho=coefs,
+            sigma=sigma,
+            constant=True,
+            dims="obs_id_fut",
+        )
+        trend = pm.Deterministic("trend_fut", alpha + beta * t_fut)
+        mu = ar1_fut + trend
+
+        yhat_fut = pm.Normal("yhat_fut", mu=mu, sigma=sigma, dims="obs_id_fut")
+        # use the updated values and predict outcomes and probabilities:
+        idata_preds = pm.sample_posterior_predictive(
+            idata_ar, var_names=["likelihood", "yhat_fut"], predictions=True, random_seed=100
+        )
+
+    return idata_ar, idata_preds, AR
+```
+
+We will fit this model by specifying priors on the negative trend and the range of the standard deviation to respect the direction of the data drift.
+
+```{code-cell} ipython3
+priors_0 = {
+    "coefs": {"mu": [0.2, 0.2], "sigma": [0.5, 0.03], "size": 2},
+    "alpha": {"mu": -4, "sigma": 0.1},
+    "beta": {"mu": -0.1, "sigma": 0.2},
+    "sigma": 8,
+    "init": {"mu": -4, "sigma": 0.1, "size": 1},
+}
+
+
+idata_trend, preds_trend, model = make_latent_AR_trend_model(y_t, priors_0, full_sample=True)
+```
+
+```{code-cell} ipython3
+pm.model_to_graphviz(model)
+```
+
+We can see the structure more clearly with the plate notation, and this additional structure has helped to appropriately fit the directional trend of the timeseries data.
+
+```{code-cell} ipython3
+plot_fits(idata_trend, preds_trend);
+```
+
+```{code-cell} ipython3
+az.summary(idata_trend, var_names=["coefs", "sigma", "alpha", "beta"])
+```
+
+## Complicating the picture further
```

Comment:
Again, just a stylistic suggestion, but I'd recommend to be more descriptive and positive about modelling more complex structure rather than arbitrarily adding complexity.

Reply:
I've made it more upbeat with remarks about adding structure to capture aspects of real world data.